core/src/runtime/state.rs

```diff
@@ -133,3 +133,16 @@ impl ExecutionState {
         Ok(())
     }
 }
+
+#[derive(Debug, Clone, Default, Copy, PartialEq, Eq)]
+pub struct EventCounts {
```

Comment:
1) why is this struct here if it's not in ExecutionState
2) just use HashMap<Opcode, usize> for most of the fields?

Reply:
Ya, I'll move it to another file and use a hashmap.

Reply:
Actually, I ended up creating a static array so that it can have the Copy trait.